ssl/quic/quic_rcidm.c

```diff
@@ -0,0 +1,693 @@
+/*
+ * Copyright 2023 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include "internal/quic_rcidm.h"
+#include "internal/priority_queue.h"
+#include "internal/list.h"
+#include "internal/common.h"
+
+/*
+ * QUIC Remote Connection ID Manager
+ * =================================
+ *
+ * We can receive an arbitrary number of RCIDs via NCID frames. Periodically, we
+ * may desire (for example for anti-connection fingerprinting reasons, etc.)
+ * to switch to a new RCID according to some arbitrary policy such as the number
+ * of packets we have sent.
+ *
+ * When we do this we should move to the next RCID in the sequence of received
+ * RCIDs ordered by sequence number. For example, if a peer sends us three NCID
+ * frames with sequence numbers 10, 11, 12, we should seek to consume these
+ * RCIDs in order.
+ *
+ * However, due to the possibility of packet reordering in the network, NCID
+ * frames might be received out of order. Thus if a peer sends us NCID frames
+ * with sequence numbers 12, 10, 11, we should still consume the RCID with
+ * sequence number 10 before consuming the RCIDs with sequence numbers 11 or 12.
```

Comment:
Why is this policy needed? I.e., why we cannot switch back to earlier RCID by sequence number? AFAIK nothing in RFC specifies this. Or am I wrong?